edge-modules/azure-monitor/tests/Microsoft.Azure.Devices.Edge.Azure.Monitor.Test/MiscellaneousTests.cs

```diff
@@ -85,6 +85,128 @@ public void TestMultipleEndpointFilter()
+
+            bigList = metrics4.Concat(metrics5);
+
+            filter = new MetricFilter("total_network_out_bytes edgehub_gettwin_total");
+            bigList = bigList.Where(x => filter.Matches(x));
+            Assert.True(bigList.Count() == 4);
+
+            // add condition on one endpoint, but not the other
+            filter = new MetricFilter("total_network_out_bytes edgehub_gettwin_total{edge_device=\"Ubuntu-21\"}");
+            bigList = bigList.Where(x => filter.Matches(x));
+            Assert.True(bigList.Count() == 3);
+        }
+
+        // Label Equality & regex tests -- added for completeness and as a safeguard against future changes (and therefore, potential bugs)
+
+        [Fact]
+        public void TestWildcard()
+        {
+            string prometheusMessage = "edgehub_gettwin_total{iothub=\"IoTHub.azure-devices.net\",edge_device=\"Ubuntu-20\",instance_number=\"b9e51990-f3f9-4d90-8fc7-ef62d01929b2\",module_name=\"edgeHub\",ms_telemetry=\"False\"} 1683388"
+                + System.Environment.NewLine
+                + "total_network_out_bytes{iothub =\"IoTHub.azure-devices.net\",edge_device=\"Ubuntu-20\",instance_number=\"b9e51990-f3f9-4d90-8fc7-ef62d01929b2\",module_name=\"edgeAgent\",ms_telemetry=\"False\"} 5134109"
+                + System.Environment.NewLine
+                + "edgehub_gettwin_total{iothub =\"IoTHub.azure-devices.net\",edge_device=\"Ubuntu-21\",instance_number=\"023163b5-1db5-41bc-b57e-4c4d1047d9c9\",id=\"DeviceId: Ubuntu-20; ModuleId: IoTEdgeMetricsCollector [IotHubHostName: IoTHub.azure-devices.net]\"} 2"
+                + System.Environment.NewLine
+                + "max_network_out_bytes{iothub =\"IoTHub.azure-devices.net\",edge_device=\"Ubuntu-20\",instance_number=\"023163b5-1db5-41bc-b57e-4c4d1047d9c9\",id=\"DeviceId: Ubuntu-20; ModuleId: $edgeHub [IotHubHostName: IoTHub.azure-devices.net]\"} 1";
+
+            MetricFilter filter = new MetricFilter("*");
+
+            IEnumerable<Metric> metrics = PrometheusMetricsParser.ParseMetrics(testTime, prometheusMessage, "http://VeryNoisyModule:9001/metrics");
+
+            IEnumerable<Metric> result = metrics.Where(x => filter.Matches(x));
+            Assert.True(result.Count() == 4);
+
+            // Wildcards * (any characters) and ? (any single character) can be used in metric names.
+            // For example, *CPU would match maxCPU and minCPU but not CPUMaximum.
+
+            filter = new MetricFilter("*_out_bytes");
+            result = metrics.Where(x => filter.Matches(x));
+            Assert.True(result.Count() == 2);
+
+            // ???CPU would match maxCPU and minCPU but not maximumCPU.
+            filter = new MetricFilter("???_network_out_bytes");
+            result = metrics.Where(x => filter.Matches(x));
+            Assert.True(result.Count() == 1);
+
+            filter = new MetricFilter("?????_network_out_bytes");

```

Comment:
Shouldn't this match `total_network_out_bytes` and `max_network_out_bytes`?  Doesn't `?` denote 0 or 1?

Reply:
The way this has been implemented according to our documentation [here](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-collect-and-transport-metrics?view=iotedge-2020-11&tabs=iothub#allow-and-disallow-lists), is that  ? denotes a single character.

Reply:
And to directly answer the question - it doesn't match max_network_out_bytes. 

![image](https://user-images.githubusercontent.com/90283547/137559220-6ad32da0-d63e-4c90-a283-dd3328c0bc72.png)